apps/meteor/server/services/ldap/service.ts

```diff
@@ -7,8 +7,8 @@ import { LDAPManager } from '../../lib/ldap/Manager';
 export class LDAPService extends ServiceClassInternal implements ILDAPService {
 	protected name = 'ldap';
 
-	async loginRequest(username: string, password: string): Promise<LDAPLoginResult> {
-		return LDAPManager.login(username, password);
+	async loginRequest(username: string, password?: string): Promise<LDAPLoginResult> {
+		return password ? LDAPManager.login(username, password) : LDAPManager.loginAuthenticatedUser(username);
```

Comment:
instead of adding this logic check here, simply make `loginAuthenticatedUser` public and call it directly.